src/runs/runs.service.ts

```diff
@@ -66,6 +66,21 @@ import { createClient } from '@/redis.js';
 import { ToolCall } from '@/tools/entities/tool-calls/tool-call.entity.js';
 import { SystemTools } from '@/tools/entities/tool-calls/system-call.entity.js';
 import { ensureRequestContextData } from '@/context.js';
+import { getProjectPrincipal } from '@/administration/helpers.js';
+import { RUNS_QUOTA_DAILY } from '@/config.js';
+
+export async function assertRunsQuota(newRuns = 1) {
+  const count = await ORM.em.getRepository(Run).count({
```

Comment:
Shouldn't we exclude CANCELLED and EXPIRED runs?

Reply:
I don't think so, both can eat GPU resources.